subsys/bluetooth/host/conn_internal.h

```diff
@@ -9,6 +9,7 @@
  */
 typedef enum __packed {
 	BT_CONN_DISCONNECTED,
+	BT_CONN_DISCONNECTED_PRIO,
```

Comment:
I wonder if we coultn't just reuse BT_CONN_DISCONNECT for this, at least that would be consistent with the state when we initiate the disconnect, or we need to distinct of remote vs locally initiated?

Reply:
For locally initiated it would be in the "DISCONNECT" already then, when receiving the HCI_DISCONNECT_COMPLETE event, so it would end up with a same state transition. `bt_conn_set_state` would return immediately in that case.